aiida/tools/importexport/dbimport/backends/common.py

```diff
@@ -108,3 +108,17 @@ def _sanitize_extras(fields: dict) -> dict:
     if fields.get('node_type', '').endswith('code.Code.'):
         fields['extras'] = {key: value for key, value in fields['extras'].items() if not key == 'hidden'}
     return fields
+
+
+def _strip_checkpoints(fields: dict) -> dict:
+    """Remove checkpoint from attributes of process nodes.
+
+    :param fields: the database fields for the entity
+    """
+    if fields.get('node_type', '').startswith('process.'):
+        fields = copy.copy(fields)
```

Comment:
Why is the copy needed here? And if it is needed, shouldn't we use `deepcopy` because we are manipulating a key inside a nested dictionary so the `fields['attribute']` will still have a reference to the original object, wouldn't it?

Reply:
> Why is the copy needed here?

TBH, I'm not sure why the copy is needed -- that is simply copy/pasted from the `_sanitize_extras` above. I don't have enough context on the whole import procedure to know if it's necessary.

>  the fields['attribute'] will still have a reference to the original object, wouldn't it

Right, good catch. Maybe a complete `deepcopy` is a bit overkill, but we could also `copy.copy` the `fields['attributes']`.

Reply:
Since we're now reconstructing the `attributes` dict, this should be resolved - but feel free to double-check @sphuber.